tcms/kiwi_auth/views.py

```diff
@@ -27,65 +29,66 @@ class PasswordResetView(views.PasswordResetView):
 
 
-@require_http_methods(['GET', 'POST'])
-def register(request):
-    """Register method of account"""
-    if request.method == 'POST':
-        form = forms.RegistrationForm(data=request.POST, files=request.FILES)
-        if form.is_valid():
-            new_user = form.save()
-            activation_key = form.set_activation_key()
-            # send a signal that new user has been registered
-            USER_REGISTERED_SIGNAL.send(sender=form.__class__,
-                                        request=request,
-                                        user=new_user)
-
-            # Send confirmation email to new user
-            if settings.DEFAULT_FROM_EMAIL and settings.AUTO_APPROVE_NEW_USERS:
-                form.send_confirm_mail(request, activation_key)
-
-                messages.add_message(
-                    request,
-                    messages.SUCCESS,
-                    _('Your account has been created, please check your mailbox for confirmation')
-                )
-            else:
-                messages.add_message(
-                    request,
-                    messages.WARNING,
-                    _('Your account has been created, but you need an administrator to activate it')
-                )
-                messages.add_message(
-                    request,
-                    messages.INFO,
-                    _('Following is the administrator list')
-                )
-
-                # super-users can approve others
-                for user in User.objects.filter(is_superuser=True):
-                    messages.add_message(
-                        request,
-                        messages.INFO,
-                        '<a href="mailto:{}">{}</a>'.format(user.email,
-                                                            user.get_full_name() or user.username)
-                    )
-
-                # site admins should be able to do so too
-                for name, email in settings.ADMINS:
-                    messages.add_message(
-                        request,
-                        messages.WARNING,
-                        '<a href="mailto:{}">{}</a>'.format(email, name)
-                    )
-
-            return HttpResponseRedirect(reverse('core-views-index'))
-    else:
-        form = forms.RegistrationForm()
-
-    context_data = {
-        'form': form,
-    }
-    return render(request, 'registration/registration_form.html', context_data)
+class Register(View):
+    """
+    Register account view.
+    """
+    template_name = 'registration/registration_form.html'
+    form_class = RegistrationForm
+    success_url = reverse_lazy('core-views-index')
+
+    def post(self, request):
+        """ Post request handler. """
+        form = self.form_class(data=request.POST, files=request.FILES)
+        if not form.is_valid():
+            self.show_invalid_messages(request)
+            return render(request, self.template_name, {'form': self.form_class()})
```

Comment:
here you pass the `form` object which contains the errors & the input data, not a new one.